packages/protocol/test/ERC1724/ZkAsset.js

```diff
@@ -1,52 +1,93 @@
 /* global artifacts, expect, contract, beforeEach, it:true */
-// ### External Dependencies
-const BN = require('bn.js');
-const crypto = require('crypto');
-
-// ### Internal Dependencies
-// eslint-disable-next-line object-curly-newline
-const { encoder, note, proof, signer } = require('aztec.js');
-const { constants, proofs } = require('@aztec/dev-utils');
+const { JoinSplitProof, note, signer } = require('aztec.js');
+const {
+    constants,
+    proofs: { JOIN_SPLIT_PROOF },
+} = require('@aztec/dev-utils');
 const secp256k1 = require('@aztec/secp256k1');
 const typedData = require('@aztec/typed-data');
+const BN = require('bn.js');
+const crypto = require('crypto');
 const truffleAssert = require('truffle-assertions');
 const { keccak256, padLeft } = require('web3-utils');
 
-const { outputCoder } = encoder;
-const { JOIN_SPLIT_PROOF } = proofs;
-
-// ### Artifacts
 const ACE = artifacts.require('./ACE');
-const JoinSplit = artifacts.require('./JoinSplit');
 const ERC20Mintable = artifacts.require('./ERC20Mintable');
-const JoinSplitInterface = artifacts.require('./JoinSplitInterface');
 const ZkAsset = artifacts.require('./ZkAsset');
+const JoinSplitValidator = artifacts.require('./JoinSplit');
+const JoinSplitValidatorInterface = artifacts.require('./JoinSplitInterface');
+JoinSplitValidator.abi = JoinSplitValidatorInterface.abi;
 
-JoinSplit.abi = JoinSplitInterface.abi;
+let joinSplitValidator;
 
 const computeDomainHash = (validatorAddress) => {
     const types = { EIP712Domain: constants.eip712.EIP712_DOMAIN };
     const domain = signer.generateZKAssetDomainParams(validatorAddress);
     return keccak256(`0x${typedData.encodeMessageData(types, 'EIP712Domain', domain)}`);
 };
 
-contract.skip('ZkAsset', (accounts) => {
+const randomAddress = () => {
+    return `0x${padLeft(crypto.randomBytes(20).toString('hex'))}`;
+};
+
+const setupSingleProofTest = async (noteValues) => {
+    const numNotes = noteValues.length;
+    const aztecAccounts = [...new Array(numNotes)].map(() => secp256k1.generateAccount());
```

Comment:
One thing that I did and was highly effective in my refactored tests was to get rid of `aztecAccounts` and notes. That is, simply have one or two accounts and a few notes that have bespoke names.

Reply:
Good ideas, refactored these setup functions and updated tests